server/controllers/finance/accounts/references.compute.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Accounts References Computations
+ */
+const Q = require('q');
+const db = require('../../../lib/db');
+
+/**
+ * @method computeAllAccountReference
+ *
+ * @description
+ * compute value of all account references and returns an array of all accounts reference
+ * with their debit, credit and balance
+ *
+ * @param {number} periodId - the period needed
+ */
+function computeAllAccountReference(periodId) {
+  const glb = {};
+
+  // get fiscal year information for the given period
+  const queryFiscalYear = `
+    SELECT fy.id, p.number AS period_number FROM fiscal_year fy
+    JOIN period p ON p.fiscal_year_id = fy.id
+    WHERE p.id = ?
+  `;
+
+  // get all references
+  const queryAccountReferences = `
+    SELECT id, abbr, description, is_amo_dep FROM account_reference;
+  `;
+
+  return db.one(queryFiscalYear, [periodId])
+    .then(fiscalYear => {
+      glb.fiscalYear = fiscalYear;
+
+      return db.exec(queryAccountReferences);
+    })
+    .then(accountReferences => {
+      const dbPromises = accountReferences.map(ar => {
+        return getValueForReference(
+          ar.abbr,
+          ar.is_amo_dep,
+          ar.description,
+          glb.fiscalYear.period_number,
+          glb.fiscalYear.id
+        );
+      });
+      return Q.all(dbPromises);
+    })
+    .then(data => data);
+}
+
+/**
+ * @method computeSingleAccountReference
+ *
+ * @description
+ * Returns the debit, credit and balance of the account reference given as an array
+ *
+ * @param {string} abbr - the reference of accounts. ex. AA or AX
+ * @param {number} periodId - the period needed
+ * @param {boolean} isAmoDep - the concerned reference is for amortissement, depreciation or provision
+ */
+function computeSingleAccountReference(abbr, isAmoDep = 0, periodId) {
+  const glb = {};
+
+  // get fiscal year information for the given period
+  const queryFiscalYear = `
+    SELECT fy.id, p.number AS period_number FROM fiscal_year fy
```

Comment:
You use the `queryFiscalYear` SQL twice.  It might be useful to make that its own function.  Like:
```js
function getFiscalYearByPeriodId(periodId) {
  const sql =  `
    SELECT fy.id, p.number AS period_number FROM fiscal_year fy
    JOIN period p ON p.fiscal_year_id = fy.id
    WHERE p.id = ?
  `;
  return db.one(sql, [periodId];
}
```

Then you could call it wherever you needed it.  Eventually we would refactor it into either `fiscal.js` or `period.js`.